astronomer/providers/snowflake/operators/snowflake.py

```diff
@@ -31,6 +38,33 @@
 from astronomer.providers.utils.typing_compat import Context
 
 
+def _check_queries_finish(conn: SnowflakeConnection, query_ids: list[str]) -> bool:
+    """Check whether snowflake queries finish (in aborting, failed_with_error or success)"""
+    with closing(conn) as conn:
+        for query_id in query_ids:
+            status = conn.get_query_status(query_id)
+
+            if status == QueryStatus.ABORTING:
+                raise AirflowException(f"ABORTING {ABORTING_MESSAGE}")
+            elif status == QueryStatus.FAILED_WITH_ERROR:
+                raise AirflowException(f"FAILED_WITH_ERROR {FAILED_WITH_ERROR_MESSAGE}")
+            elif status == QueryStatus.SUCCESS:
+                pass
+            elif status in (
+                QueryStatus.RUNNING,
+                QueryStatus.QUEUED,
+                QueryStatus.DISCONNECTED,
```

Comment:
Should QueryStatus.DISCONNECTED also qualify for an exception? Is Snowflake automatically able to resolve disconnected queries? If not, we might keep on polling for it until triggerer times out.

Reply:
I kinda hesitate on `QueryStatus.DISCONNECTED`. According to [doc](https://docs.snowflake.com/developer-guide/python-connector/python-connector-api#enums), `The session’s connection is broken. The query’s state will change to “FAILED_WITH_ERROR” soon.`, it's going to failed but just not yet failed.

Reply:
Makes sense, but if it is anyway going to change to Failed with error then can we not raise an early exception? 

I am happy with whatever you feel is the best. Just had this thought.

Reply:
After a second thought, I think your idea is better. After I applied your suggestion, I added a comment there cause it might still be a bit confusing.